src/sortStudents.ts

```diff
@@ -1,16 +1,69 @@
 
 export interface Student {
-  // describe Student interface
+  name:string;
+  surname:string;
+  age:number;
+  married:boolean;
+  grades:number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age ='age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function averCal(arr: number[]): number {
+  const average = arr.reduce((a, b) => a + b, 0) / arr.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return average;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const newStudent: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        newStudent
+          .sort((a: Student, b:Student) => a[sortBy].localeCompare(b[sortBy]));
+      }
+
+      if (order === 'desc') {
+        newStudent
+          .sort((a:Student, b:Student) => b[sortBy].localeCompare(a[sortBy]));
+      }
```

Comment:
```suggestion
        newStudent
          .sort((a: Student, b:Student) => order === 'asc'
            ? a[sortBy].localeCompare(b[sortBy])
            : b[sortBy].localeCompare(a[sortBy])
```

Good case to use ternary operator